File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    private double sideLen;

    public Square(double sideLen) {
        if(sideLen <= 0) throw new IllegalArgumentException("Invalid Sidelen: sidelen must be positive");
        this.sideLen = sideLen;
    }
    
    //setter and getter
    public double getSidelen(){
        return sideLen;
    }

    public void setSidelen(double sl){
        this.sideLen = sl;
    }
    
    @Override
    public double getArea() {
        return sideLen * sideLen;
    }
}

File: typescript/src/shapes/square.ts
function newsquare(sideLen: number): Shape {
    if (sideLen <= 0) {
        throw new Error('Side length must be positive');
    }

    return {
        computeArea: () => sideLen * sideLen,
        getName: () => 'Square'
    };
}

export { newsquare }